payjoin/src/receive/v2/mod.rs

```diff
@@ -74,13 +91,14 @@ impl Receiver {
     ///
     /// # References
     /// - [BIP 77: Payjoin Version 2: Serverless Payjoin](https://github.com/bitcoin/bips/pull/1483)
+    #[allow(clippy::new_ret_no_self)]
```

Comment:
This warning should be heeded. It suggests this constructor belongs on `EphemeralReceiver`